### DISCUSSION TITLE
v2.0 diarization processing time

### DISCUSSION THREAD

**Discussion by martinjbaker** [pyannote/pyannote-audio#852] [Q&A]
I'm seeing way longer processing times with v2 compared with v1.

Same GPU hardware and same source file takes 3x longer to process with v2. I can't deny that the accuracy is improved but is this something that is expected to change or is it just an inevitable tradeoff for better accuracy?

**Comment by hbredin** [comment]
Thanks for your interest in `pyannote.audio`.

Switching to the (much larger) `speechbrain` speaker embedding is the main reason of this slow down (and for half of the accuracy improvement).

I can think of many ways of making things faster but this is definitely not my priority right now, especially since I am still working on the speaker diarization pipeline internals, trying to improve accuracy even more.

**Reply by pengyuan101** [reply]
Thanks for your work!  it really helped me a lot ,  but can you give me some suggestions for speeding up?

**Reply by hbredin** [reply]
> I can think of many ways of making things faster but this is definitely not my priority right now, especially since I am still working on the speaker diarization pipeline internals, trying to improve accuracy even more.

[Latest commit](https://github.com/pyannote/pyannote-audio/commit/cd8b8b4a4a029edcdb6c349efe2580c2945c33cc) in `develop` branch brings a 35% speed up when applied to AMI test set on one GPU (from 40min down to 26min).

**Reply by RYucel** [reply]
Sorry for noob question. How can I use develop version if available? I still have a very slow process for diarization. Any improvement welcome

**Comment by CompRhys** [comment]
When we use the pre-trained models on the AMI data set are their issues arising from the fact that we don't first break up the long meetings ~35 minutes into shorter segments for the model to process? I can't find any detailed documentation on the pre-trained model to work out whether/how they break up longer sequences.

**Comment by RezaBi** [comment]
Hi, is it possible to run diarization model on pyannote v2 on cpus?? I test some files and never got to end of it!

**Reply by RezaBi** [reply]
@hbredin

**Reply by hbredin** [reply]
Yes.
